sdk/core/core-https/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -34,26 +73,73 @@ export function bearerTokenAuthenticationPolicy(
 ): PipelinePolicy {
   const { credential, scopes } = options;
   const tokenCache: AccessTokenCache = new ExpiringAccessTokenCache();
-  async function getToken(tokenOptions: GetTokenOptions): Promise<string | undefined> {
+  const { prepareRequest, getChallenge = defaultGetChallenge, processChallenge } =
+    options.challenge || {};
+
+  /**
+   * getToken will call to the underlying credential's getToken request with properties coming from the request,
+   * as well as properties coming from parsing the CAE challenge, if any.
+   */
+  async function getToken(
+    request: PipelineRequest,
+    context?: AuthenticationContext
+  ): Promise<string | undefined> {
     let accessToken = tokenCache.getCachedToken();
+    const getTokenOptions: GetTokenOptions = {
+      claims: context?.claims,
+      abortSignal: request.abortSignal,
+      tracingOptions: {
+        spanOptions: request.spanOptions
+      }
+    };
     if (accessToken === undefined) {
-      accessToken = (await credential.getToken(scopes, tokenOptions)) || undefined;
+      accessToken =
+        (await credential.getToken(context?.scopes || scopes, getTokenOptions)) || undefined;
       tokenCache.setCachedToken(accessToken);
     }
-
     return accessToken ? accessToken.token : undefined;
   }
+
+  /**
+   * Populates the token in the request headers.
+   */
+  function assignToken(request: PipelineRequest, token?: string): PipelineRequest {
```

Comment:
feels like either token shouldn't be optional or we shouldn't set the header if it's not defined